### PR TITLE
Everything for Impl Phase 2

### PR DESCRIPTION
**Pull request by jpmartin2**
#### What issue does this fix?
- #68 

There are a lot of changes here.

### CODE REVIEW

File: src/main/java/aeneas/views/BoardView.java [hunk 1]
```diff
@@ -41,11 +41,12 @@
 
   SquareView[][] grid = new SquareView[Board.SIZE][Board.SIZE];
   Board board;
+  Model gameModel;
   private int dragDropRow = 0, dragDropCol = 0;
   private SquareClickListener clickListener;
   private SquareDragListener dragListener;
   private SquareDropListener dropListener;
-  private PlacedPiece pieceBeingDragged;
+  private PlacedPiece pieceBeingDragged = null;
```
Comment (PeterMitrano) [inline src/main/java/aeneas/views/BoardView.java:54]: Uhhh why?

**Comment by PeterMitrano** [comment]
Good call on the setting success. :+1:

**Comment by garrisonhefter** [comment]
When I run this branch I get null-pointer exceptions every time I try to click on a square to toggle it on/off in the builder?

**Comment by jpmartin2** [comment]
Yeah, Peter's drag and drop stuff broke that. I'll have a fix soon

**Comment by jkuszmaul** [comment]
The tests are fixed in #70

**Comment by PeterMitrano** [comment]
So does #70 replace this? Do we still need this

**Comment by jpmartin2** [comment]
Yeah, #70 has this and more.

**Pull request closed without merging**